packages/solana/src/client.ts

```diff
@@ -426,44 +461,40 @@ export class Web3Modal extends Web3ModalScaffold {
   public async switchNetwork(caipNetwork: CaipNetwork) {
     const caipChainId = caipNetwork.id
     const providerType = SolStoreUtil.state.providerType
+    const provider = SolStoreUtil.state.provider
     const chain = SolHelpersUtil.getChainFromCaip(this.chains, caipChainId)
 
-    if (this.chains) {
-      if (chain) {
-        SolStoreUtil.setCaipChainId(`solana:${chain.chainId}`)
-        SolStoreUtil.setCurrentChain(chain)
-        localStorage.setItem(SolConstantsUtil.CAIP_CHAIN_ID, `solana:${chain.chainId}`)
-        if (providerType?.includes(ConstantsUtil.INJECTED_CONNECTOR_ID)) {
-          SolStoreUtil.setConnection(
-            new Connection(
-              SolHelpersUtil.detectRpcUrl(chain, OptionsController.state.projectId),
-              this.connectionSettings
-            )
-          )
-          this.setAddress(
-            this.walletAdapters
-              .find(
-                adapter => adapter.name.toLocaleLowerCase() === providerType.toLocaleLowerCase()
-              )
-              ?.publicKey?.toString()
+    if (chain) {
```

Comment:
@glitch-txs I think `this.chain` is not same with the `chain` that returned from `getChainFromCaip`. `this.chain` should be used for a reference to `solana` to indicate what chain is this client. See the following line:
```
private chain: AvailableChain = CommonConstantsUtil.CHAIN.SOLANA
```

Reply:
Here it is used in SolStoreUtil, which does not rewrite this.chain, and stores data about chainId, explorer, etc.
while, as I understood this.chain tells 'EVM' or 'solana' in use. Probably we should rename it to "mode"?

Reply:
@phoenixVS `mode` wouldn't be a good naming IMO. Solana and EVMs are chains while the Solana Mainnet / Solana DevNet / Polygon etc are networks. 

The `chain` should be actually used for higher scope values; `Ethereum`, `Solana`, `Polkadot`, `Bitcoin` etc.